### ISSUE TITLE
Missing Pixel-to-Physical Unit Conversion in Radial Profile Plotting

### ISSUE THREAD

**Issue by SunnySco** [Autostronomy/AstroPhot#244]
### Description:
I’ve encountered a bug in the `radial_median_profile` function. When plotting the surface brightness profile, the radius values are not being converted from pixels to physical units (such as arcseconds). As a result, the radial bins and the x-axis labels are in pixel units instead of the intended physical units. This would cause missing data points at further distance. 

**Steps to Reproduce:**
1. Run the `radial_median_profile` function.
2. Notice that the x-axis of the plot is labeled with the radial units specified by `rad_unit` (e.g., "arcsec"), but the radius values are not converted from pixels to the specified physical unit.

Here's the source code:
```
def radial_median_profile(
    fig,
    ax,
    model: "AstroPhot_Model",
    count_limit: int = 10,
    return_profile: bool = False,
    rad_unit: str = "arcsec",
    doassert: bool = True,
    plot_kwargs: dict = {},
):
    """Plot an SB profile by taking flux median at each radius.

    Using the coordinate transforms defined by the model object,
    assigns a radius to each pixel then bins the pixel-radii and
    computes the median in each bin. This gives a simple
    representation of the image data if one were to simply average the
    pixels along isophotes.

    Args:
      fig: matplotlib figure object
      ax: matplotlib axis object
      model (AstroPhot_Model): Model object from which to determine the radial binning. Also provides the target image to extract the data
      count_limit (int): The limit of pixels in a bin, below which uncertainties are not computed. Default: 10
      return_profile (bool): Instead of just returning the fig and ax object, will return the extracted profile formatted as: Rbins (the radial bin edges), medians (the median in each bin), scatter (the 16-84 quartile range / 2), count (the number of pixels in each bin). Default: False
      rad_unit (str): The name of the physical radius units. Default: "arcsec"
      doassert (bool): If any requirements are imposed on which kind of profile can be plotted, this activates them. Default: True

    """

    Rlast_phys = torch.max(model.window.shape / 2).item()
    Rlast_pix = Rlast_phys / model.target.pixel_length.item()

    Rbins = [0.0]
    while Rbins[-1] < Rlast_pix:
        Rbins.append(Rbins[-1] + max(2, Rbins[-1] * 0.1))
    Rbins = np.array(Rbins)

    with torch.no_grad():
        image = model.target[model.window]
        X, Y = image.get_coordinate_meshgrid() - model["center"].value[..., None, None]
        X, Y = model.transform_coordinates(X, Y)
        R = model.radius_metric(X, Y)
        R = R.detach().cpu().numpy()

    count, bins, binnum = binned_statistic(
        R.ravel(),
        image.data.detach().cpu().numpy().ravel(),
        statistic="count",
        bins=Rbins,
    )

    stat, bins, binnum = binned_statistic(
        R.ravel(),
        image.data.detach().cpu().numpy().ravel(),
        statistic="median",
        bins=Rbins,
    )
    stat[count == 0] = np.nan

    scat, bins, binnum = binned_statistic(
        R.ravel(),
        image.data.detach().cpu().numpy().ravel(),
        statistic=partial(iqr, rng=(16, 84)),
        bins=Rbins,
    )
    scat[count > count_limit] /= 2 * np.sqrt(count[count > count_limit])
    scat[count <= count_limit] = 0

    if model.target.zeropoint is not None:
        stat = flux_to_sb(stat, model.target.pixel_area.item(), model.target.zeropoint.item())
        ax.set_ylabel("Surface Brightness [mag/arcsec$^2$]")
        if not ax.yaxis_inverted():
            ax.invert_yaxis()
    else:
        stat = np.log10(stat)
        ax.set_ylabel("log$_{10}$(flux/arcsec^2)")

    kwargs = {
        "linewidth": 0,
        "elinewidth": 1,
        "color": main_pallet["primary2"],
        "label": "data profile",
    }
    kwargs.update(plot_kwargs)
    ax.errorbar(
        (Rbins[:-1] + Rbins[1:]) / 2,
        stat,
        yerr=scat,
        fmt=".",
        **kwargs,
    )
    ax.set_xlabel(f"Radius [{rad_unit}]")

    if return_profile:
        return Rbins, stat, scat, count
    return fig, ax
```

**Comment by ConnorStoneAstro** [comment]
Hmmm, I think I see what you mean. It is this section:
```python
Rlast_pix = Rlast_phys / model.target.pixel_length.item()

Rbins = [0.0]
while Rbins[-1] < Rlast_pix:
    Rbins.append(Rbins[-1] + max(2, Rbins[-1] * 0.1))
Rbins = np.array(Rbins)
```
correct?

It looks like the bins are set in pixel units though the rest of the calculations are all done in physical units. I must have done my testing with a pixelscale near 1! I'll see if I can fix this right away!

**Comment by ConnorStoneAstro** [comment]
Ok I've implemented the fix. It will need to go through a bunch of automated testing, like all PRs, but the patch should be pushed to pip either tonight or early tomorrow! Thanks so much for spotting this! :)

**Comment by ConnorStoneAstro** [comment]
Hi @SunnySco Looks like everything went through the checks and the bug fix is now online in version 0.16.8 :) Thanks for spotting this, careful users like you make AstroPhot better for everyone! Certainly feel free to let me know if anything else comes up. 
- Connor

**Comment by SunnySco** [comment]
Thanks for the quick fix! Glad I could help improve AstroPhot. Definitely a very helpful tool!
